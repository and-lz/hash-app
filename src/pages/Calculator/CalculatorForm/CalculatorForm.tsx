import React, { useRef, useState } from 'react'
import Button from 'components/Button/Button'
import { Form } from 'design-system/Form/Form'
import { VerticalSpacer } from 'design-system/Spaces/Spaces'
import Input from 'components/Input/Input'
import InputCurrency from 'components/InputCurrency/InputCurrency'

interface CalculadoraFormProps {
  onSubmit: Function
}

function CalculatorForm({ onSubmit }: CalculadoraFormProps) {
  const form = (useRef() as unknown) as HTMLFormElement

  const [amount, setAmount] = useState(1000)
  const [installments, setInstallments] = useState(12)
  const [mdr, setMdr] = useState(1)

  function checkFormValidity() {
    if (form.current.checkValidity()) {
      onSubmit(amount, installments, mdr)
    }
  }

  return (
    <Form
      ref={form}
      onSubmit={(e: React.FormEvent<HTMLInputElement>) => {
        e.preventDefault()
        checkFormValidity()
      }}
    >
      <InputCurrency
        autofocus
        onChange={(value: number) => setAmount(value)}
        required
        label="Informe o valor da renda"
        value={amount}
        min={1000}
        max={100000000}
      />
      <VerticalSpacer />
      <Input
        onChange={(value: number) => setInstallments(value)}
        required
        label="Em quantas parcelas"
        value={installments}
        addon="MÃ¡ximo de 12 parcelas"
        type="number"
        min={1}
        max={12}
      />
      <VerticalSpacer />
      <Input
        onChange={(value: number) => setMdr(value)}
        required
        label="Informe o percentual de MDR"
        value={mdr}
        type="number"
        min={0}
      />
      <VerticalSpacer />
      <Button onClick={checkFormValidity} type="submit" label="Consultar" />
    </Form>
  )
}

export default CalculatorForm
