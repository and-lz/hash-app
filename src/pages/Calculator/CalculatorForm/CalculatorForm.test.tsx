import React from 'react'
import { render, RenderResult, screen } from '@testing-library/react'
import CalculatorForm from './CalculatorForm'

describe('<CalculatorForm />', () => {
  let component: RenderResult
  beforeEach(() => {
    component = render(<CalculatorForm onSubmit={() => {}} />)
  })
  test('Should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
