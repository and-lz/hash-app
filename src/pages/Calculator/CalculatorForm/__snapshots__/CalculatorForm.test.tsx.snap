// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatorForm /> Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-gsDJrp"
      >
        <label
          class="sc-eCImvq iMeBec"
          for="Informe o valor da renda field"
        >
          Informe o valor da renda
           
          *
        </label>
        <input
          class="sc-hKwCoD fWjqww"
          data-testid="input-field"
          id="Informe o valor da renda field"
          required=""
          type="text"
          value="R$ 1.000,00"
        />
        <hr
          class="sc-dkPtyc icWbJV"
        />
        <label
          class="sc-eCImvq iMeBec"
          for="Em quantas parcelas field"
        >
          Em quantas parcelas
           
          *
        </label>
        <input
          class="sc-hKwCoD fWjqww"
          data-testid="input-field"
          id="Em quantas parcelas field"
          max="12"
          min="1"
          required=""
          type="number"
          value="12"
        />
        <div
          class="sc-jRQAMF VyLUY"
        >
          Máximo de 12 parcelas
        </div>
        <hr
          class="sc-dkPtyc icWbJV"
        />
        <label
          class="sc-eCImvq iMeBec"
          for="Informe o percentual de MDR field"
        >
          Informe o percentual de MDR
           
          *
        </label>
        <input
          class="sc-hKwCoD fWjqww"
          data-testid="input-field"
          id="Informe o percentual de MDR field"
          min="0"
          required=""
          type="number"
          value="1"
        />
        <hr
          class="sc-dkPtyc icWbJV"
        />
        <button
          class="sc-bdvvaa ihLWWn"
        >
          Consultar
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-gsDJrp"
    >
      <label
        class="sc-eCImvq iMeBec"
        for="Informe o valor da renda field"
      >
        Informe o valor da renda
         
        *
      </label>
      <input
        class="sc-hKwCoD fWjqww"
        data-testid="input-field"
        id="Informe o valor da renda field"
        required=""
        type="text"
        value="R$ 1.000,00"
      />
      <hr
        class="sc-dkPtyc icWbJV"
      />
      <label
        class="sc-eCImvq iMeBec"
        for="Em quantas parcelas field"
      >
        Em quantas parcelas
         
        *
      </label>
      <input
        class="sc-hKwCoD fWjqww"
        data-testid="input-field"
        id="Em quantas parcelas field"
        max="12"
        min="1"
        required=""
        type="number"
        value="12"
      />
      <div
        class="sc-jRQAMF VyLUY"
      >
        Máximo de 12 parcelas
      </div>
      <hr
        class="sc-dkPtyc icWbJV"
      />
      <label
        class="sc-eCImvq iMeBec"
        for="Informe o percentual de MDR field"
      >
        Informe o percentual de MDR
         
        *
      </label>
      <input
        class="sc-hKwCoD fWjqww"
        data-testid="input-field"
        id="Informe o percentual de MDR field"
        min="0"
        required=""
        type="number"
        value="1"
      />
      <hr
        class="sc-dkPtyc icWbJV"
      />
      <button
        class="sc-bdvvaa ihLWWn"
      >
        Consultar
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
